### PR TITLE
Fix RUBY-2055 Driver sends null pwd field in createUser when password is not specified

### PR DESCRIPTION
**Pull request by p-mongo**
https://jira.mongodb.com/browse/RUBY-2055

### CODE REVIEW

File: docs/tutorials/user-management.txt [hunk 1]
```diff
@@ -34,12 +37,39 @@ In this example, all operations would be performed on the ``users`` database.
...
+
+  client.use('$external').database.users.create(
+    'C=US,ST=New York,L=New York City,O=MongoDB,OU=x509,CN=localhost',
+    roles: [{role: 'read', db: 'admin'}],
+  )
+
+Note that the auth source is not specified for creating the user - auth source
+is only used during the authentication process. If ``#create`` is invoked with
+a ``User`` object with ``auth_source`` set, the auth source is ignored for
+the purposes of user management.
```
Comment (egiurleo) [inline docs/tutorials/user-management.txt]: Could you mention that the user's auth source will be set to a default value depending on what mechanism you use to create the user?

Comment (p-mongo) [reply docs/tutorials/user-management.txt]: I suppose the language could be made more clear but my current understanding is that "auth source" is something that only exists during authentication. When performing user management there is no concept of auth source at all.

Comment (p-mongo) [reply docs/tutorials/user-management.txt]: So, to respond to your comment:

> the user's auth source will be set to a default value

Set *where* exactly?

Comment (egiurleo) [reply docs/tutorials/user-management.txt]: Having looked at the code a bit more, I agree with you. Thanks for pointing that out. Merge away.

**Pull request merged**